CrewChiefV4/Events/LapTimes.cs

```diff
@@ -363,30 +359,49 @@ protected override void triggerInternal(GameStateData previousGameState, GameSta
                     (currentGameState.OpponentData.Count == 0 || (currentGameState.OpponentData.Count == 1 && currentGameState.OpponentData.First().Value.DriverRawName == currentGameState.SessionData.DriverRawName));
                 if (isHotLappingOrLonePractice)
                 {
+                    // TODO: self pace stuff is still overall, not conditions dependent
                     lapAndSectorsComparisonData[0] = lapAndSectorsSelfComparisonData[0];
                     lapAndSectorsComparisonData[1] = lapAndSectorsSelfComparisonData[1];
                     lapAndSectorsComparisonData[2] = lapAndSectorsSelfComparisonData[2];
                     lapAndSectorsComparisonData[3] = lapAndSectorsSelfComparisonData[3];
                 }
                 else
                 {
-                    if (currentGameState.SessionData.SessionType == SessionType.Race)
+                    // in qual sessions we want absolute timings. We can also use absolute timings if the conditions are static.
+                    // If the conditions are changing we want timings relative to the prevailing conditions for non-qual sessions.
+                    // For race sessions we want the recent pace
+                    TimingData.ConditionsEnum conditionsForLapComparisonData = TimingData.ConditionsEnum.CURRENT;
+                    if (currentGameState.SessionData.SessionType == SessionType.Qualify || !currentGameState.TimingData.conditionsHaveChanged)
                     {
+                        conditionsForLapComparisonData = TimingData.ConditionsEnum.ANY;
+                    }
+                    if (currentGameState.SessionData.SessionType == SessionType.Race && !currentGameState.TimingData.conditionsHaveChanged)
+                    {
+                        // use recent data
                         lapAndSectorsComparisonData = currentGameState.getTimeAndSectorsForBestOpponentLapInWindow(paceCheckLapsWindowForRaceToUse, currentGameState.carClass);
-                        Console.WriteLine("Opponents best sectors = " + lapAndSectorsComparisonData[1].ToString("0.000") + ",    " + lapAndSectorsComparisonData[2].ToString("0.000") + ",    " + lapAndSectorsComparisonData[3].ToString("0.000"));
-                        Console.WriteLine("Player best sectors = " + currentGameState.SessionData.PlayerBestSector1Time.ToString("0.000") + ",    " + currentGameState.SessionData.PlayerBestSector2Time.ToString("0.000") + ",    " + currentGameState.SessionData.PlayerBestSector3Time.ToString("0.000"));
                     }
-                    else if (currentGameState.SessionData.SessionType == SessionType.Qualify || currentGameState.SessionData.SessionType == SessionType.Practice)
+                    else
                     {
-                        lapAndSectorsComparisonData = currentGameState.getTimeAndSectorsForBestOpponentLapInWindow(-1, currentGameState.carClass);
```

Comment:
So, this function call getTimeAndSectorsForBestOpponentLapInWindow(-1, currentGam 

used to take care of a situation when best time already belongs to the player.   If we are currently starting the new lap, and we beat existing best time, we need to ignore that time (otherwise delta is 0.0) does that make sense?